File: SpotifyAPI/Models/Playlist.cs
using Microsoft.EntityFrameworkCore;

namespace Models;

[PrimaryKey("Id")]
public class Playlist
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; }
    public User Owner { get; set; }
    public PlaylistType Type { get; set; }
    public List<Track> Tracks = new();

    public Playlist(string name, User owner, PlaylistType type)
    {
        Name = name;
        Owner = owner;
        Type = type;
    }

    public Playlist()
    {
    }
}

File: spotik_mobile/lib/models/Playlist/playlist.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:spotik_mobile/models/Track/track.dart';

part 'playlist.freezed.dart';
part 'playlist.g.dart';

@freezed
class Playlist with _$Playlist{
  // ignore: invalid_annotation_target
  @JsonSerializable(explicitToJson: true)
  const factory Playlist({required String id,required String prewiewId, required String name, required String owner, required List<Track> tracks }) = _Playlist;

  factory Playlist.fromJson(Map<String, dynamic> json) => _$PlaylistFromJson(json);
}